jax/numpy/lax_numpy.py

```diff
@@ -3788,6 +3789,39 @@ def _quantile(a, q, axis, interpolation, keepdims):
   return lax.convert_element_type(result, a.dtype)
 
 
+@partial(jit, static_argnums=2)
+@partial(vmap, in_axes=(None, 0, None))
```

Comment:
The strategy of `ravel -> vmap -> reshape` is very clean, but especially on TPUs the reshapes can be sometimes be expensive.

I wonder if this might be a good use for [`vectorize`](https://jax.readthedocs.io/en/latest/jax.html#jax.numpy.vectorize) to automatically apply `vmap` instead? e.g.,
```suggestion
@partial(vectorize, excluded={0, 2})
```

This would let you eliminate the calls to `reshape` and the fewer reshapes might be more efficient.

Reply:
Done.